python/ray/serve/_private/common.py

```diff
@@ -677,6 +677,7 @@ class DeploymentHandleSource(str, Enum):
 @dataclass
 class RequestMetadata:
     request_id: str
+    internal_request_id: str
```

Comment:
qq: is the `internal_request_id` send to the app behind ray serve proxy?

Reply:
No it's not, this will appear in the serve context object during the request if there is a need for the deployment to know about it, but the request client won't have this info. Do you think there is a need to pass this back to the client? Can make a new header for it if there is a need for this.

Reply:
Oh actually reading it again, since it's in the context object, the "app" can get it through calling `ray.serve.context._serve_request_context.get(). internal_request_id`. It's not passed as a header to the "app" nor the request "client".

Reply:
^ not a public API

this internal request ID is purely an implementation detail, shouldn't be exposed to end users

Reply:
oh ok, let me take it out👍